File: Leetcode-numbers/26.c

int removeDuplicates(int* nums, int numsSize) {
    int left = 0, right = 0;
    int n = 0;
    if(numsSize == 1)
    return 1;
    while(right < numsSize-1)
    {
        if(nums[right] != nums[right+1])
        {
            n = right;
            nums[left++] = nums[right++];
            printf("%d ",nums[left-1]);
        }
        else
        right++;
    }
    if((nums[n] != nums[right]) || (nums[right] != nums[right-1]))
    nums[left] = nums[right];
    return left+1;
}

File: 150/26.go
func removeDuplicates(nums []int) int {
    left, right := 0, 1
    num := 1
    len := len(nums)
    if len == 0 {
        return 0
    } else if len == 1 {
        return 1
    }
    for right < len {
        if nums[left] == nums[right] {
            right++
        } else {
            left++
            nums[left] = nums[right]
            right++
            num++
        }
    }
    return num
}